module/zfs/dbuf.c

```diff
@@ -613,8 +614,7 @@ dbuf_cache_multilist_index_func(multilist_t *ml, void *obj)
 static inline unsigned long
 dbuf_cache_target_bytes(void)
 {
-	return MIN(dbuf_cache_max_bytes,
-	    arc_target_bytes() >> dbuf_cache_shift);
+	return MIN(dbuf_cache_max_bytes, arc_c_max >> dbuf_cache_shift);
```

Comment:
This seems like it would still defeat any manual increase to the `dbuf_cache_max_bytes` module parameter.  Is that the intention?  If so then it seems like we should initialize dbuf_cache_max_bytes to ULONG_MAX, and then we don't need to do anything with it in `dbuf_init()` (it can be tuned down from there and it will take effect here in dbuf_cache_target_bytes()).

Reply:
How about `MIN(dbuf_cache_max_bytes, arc_c_max)`? I think the intent here is to allow this to be tunable, but not allow for an unreasonable value.

Reply:
But on the other hand, initializing `dbuf_cache_max_bytes` to ULONG_MAX makes the dbuf cache target follow changes to `arc_c_max` (via `zfs_arc_max`).

Reply:
There is not much difference between setting dbuf_cache_max_bytes to ULONG_MAX and the change below.  Setting it to ULONG_MAX just leaves the question about dbuf_metadata_cache_max_bytes. I remember some requests to remove the dbuf metadata limit, but I don't remember motivation why it was added to vote for removal.

This particular chunk either way I'd leave as it is, as I have told.